### DISCUSSION TITLE
Scaphandre v1.0.0

### DISCUSSION THREAD

**Discussion by bpetit** [hubblo-org/scaphandre#357] [General]
Release v1.0.0 is out ! 🥳 

Feel free to ask questions in this discussion regarding the release and/or any comments.

Please consider [sponsoring](https://github.com/sponsors/hubblo-org) if you want this project to thrive ! 🙏🏽 

### Added

- Host resources consumption metrics : scaph_host_swap_total_bytes, scaph_host_swap_free_bytes, scaph_host_memory_free_bytes, scaph_host_memory_available_bytes, scaph_host_memory_total_bytes, scaph_host_disk_total_bytes, scaph_host_disk_available_bytes, scaph_host_cpu_frequency, scaph_host_load_avg_fifteen, scaph_host_load_avg_five, scaph_host_load_avg_one - see https://hubblo-org.github.io/scaphandre-documentation/references/metrics.html for details, https://github.com/hubblo-org/scaphandre/issues/271 and https://github.com/hubblo-org/scaphandre/pull/278 for reference
- Per-process resource consumption metrics : scaph_process_cpu_usage_percentage, scaph_process_memory_bytes, scaph_process_memory_virtual_bytes, scaph_process_disk_total_write_bytes, scaph_process_disk_write_bytes, scaph_process_disk_read_bytes, scaph_process_disk_total_read_bytes - see https://hubblo-org.github.io/scaphandre-documentation/references/metrics.html for details, https://github.com/hubblo-org/scaphandre/issues/141 and https://github.com/hubblo-org/scaphandre/pull/274 for reference
- Added service monitor to helm chart, see https://github.com/hubblo-org/scaphandre/pull/230, thanks @mmadoo
- Added packaging folder with sample systemd services files, see https://github.com/hubblo-org/scaphandre/pull/317 and https://github.com/hubblo-org/scaphandre/issues/261, thanks @jcaesar
- Added prometheus push mode exporter, see https://github.com/hubblo-org/scaphandre/issues/269
- Added RPM build github action workflow, see https://github.com/hubblo-org/scaphandre/issues/310
- Added RAPL mmio metric, when the domain is present, see https://github.com/hubblo-org/scaphandre/issues/318 and https://github.com/hubblo-org/scaphandre/pull/329
- Added specific RAPL PSYS metric, when available, see https://github.com/hubblo-org/scaphandre/issues/316 and https://github.com/hubblo-org/scaphandre/pull/329
- Filtering per process in JSON exporter, see https://github.com/hubblo-org/scaphandre/issues/216
- Github action workflow to build Windows EXE installer on each release, see https://github.com/hubblo-org/scaphandre/pull/333
- Github action workflow to build DEB package on each release, see https://github.com/hubblo-org/scaphandre/pull/352, thanks @bdromard
- Added warning messages when powercap files permissions won't allow Scaphandre to read RAPL data, see https://github.com/hubblo-org/scaphandre/issues/214

### Changed

- Global power metrics have changed and could give higher numbers than previously. Please have a look at the [documentation](https://hubblo-org.github.io/scaphandre-documentation/explanations/host_metrics.html).
- `scaph_self_mem_total_program_size`, `scaph_self_mem_resident_set_size` and `scaph_self_mem_shared_resident_size` are replaced by `scaph_self_memory_bytes` and `scaph_self_memory_virtual_bytes`, see https://github.com/hubblo-org/scaphandre/pull/274/files
- Refactored warp10 exporter, see https://github.com/hubblo-org/scaphandre/pull/291 and https://github.com/hubblo-org/scaphandre/issues/105
- Refactored exporters creation with clap4, see https://github.com/hubblo-org/scaphandre/pull/292, thanks @TheElectronWill
- Default docker-compose sets a privileged container now, otherwise it doesn't work in an apparmor context, see https://github.com/hubblo-org/scaphandre/issues/135 and https://github.com/hubblo-org/scaphandre/commit/a1a06ea280b8e66067b2c3b73ac08a377604eb61
- Moved from procfs, to sysinfo, see https://github.com/hubblo-org/scaphandre/issues/267

### Fixed

- Fixed doc broken links, see https://github.com/hubblo-org/scaphandre/pull/259 and https://github.com/hubblo-org/scaphandre/issues/288, thanks @homersimpsons
- Now works on more than 1 vcpu Qemu/KVM virtual machines, see https://github.com/hubblo-org/scaphandre/issues/133 and https://github.com/hubblo-org/scaphandre/pull/207, thanks @tawalaya
- Fix for Kubernetes, don't create PSP if version of kubernetes is above 1.25, see https://github.com/hubblo-org/scaphandre/pull/250, thanks @rossf7
- Fixed bug in --containers flag, see https://github.com/hubblo-org/scaphandre/pull/326, thanks rossf7
- Fix on qemu exporter, see https://github.com/hubblo-org/scaphandre/issues/260
- Fixed panics on regex filters, see https://github.com/hubblo-org/scaphandre/issues/295
- Fixed invalid escape sequence in Prometheus exporter, see https://github.com/hubblo-org/scaphandre/issues/204, thanks @demeringo
- Removed broken python bindings, until it is fixed, see https://github.com/hubblo-org/scaphandre/pull/315 and https://github.com/hubblo-org/scaphandre/issues/296

<hr /><em>This discussion was created from the release <a href='https://github.com/hubblo-org/scaphandre/releases/tag/v1.0.0'>Scaphandre v1.0.0</a>.</em>

**Comment by bpetit** [comment]
A special thank you to @Vic063 for his contributions on the windows driver, thus enabling Scaphandre on windows ! :)

**Comment by ejhusom** [comment]
Hi! :)

When I'm downloading the .deb package and installing using apt or dpkg on Ubuntu 22.10, the version I get is 0.5.0. Anyone else struggling with this?

Thank you for the amazing work on Scaphandre! It's an invaluable tool :)

**Reply by jcaesar** [reply]
You're in fact getting 1.0.0, just with the wrong version tag, which was forgotten and only added after 1.0.0 was released. https://github.com/hubblo-org/scaphandre/compare/c43d72b43ca5203d165db7fbff22ef2215743dd6...eebb06b6ea0d5209c34f88d7a7376a3205305675

Imho, this would warrant a 1.0.1, but in practice, you can just ignore it.

(For the AUR build, I decided to [patch](https://aur.archlinux.org/cgit/aur.git/tree/version-fix.diff?h=scaphandre)…)

**Reply by ejhusom** [reply]
I see, thanks for clearing that up! :)
